gcloud-java-pubsub/src/test/java/com/google/cloud/pubsub/AckDeadlineRenewerTest.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsub;
+
+import com.google.cloud.GrpcServiceOptions.ExecutorFactory;
+import com.google.cloud.ServiceOptions.Clock;
+import com.google.common.collect.ImmutableList;
+
+import org.easymock.Capture;
+import org.easymock.EasyMock;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+public class AckDeadlineRenewerTest {
+
+  private static final int MIN_DEADLINE_MILLISECONDS = 10_000;
+
+  private static final String SUBSCRIPTION1 = "subscription1";
+  private static final String SUBSCRIPTION2 = "subscription2";
+  private static final String ACK_ID1 = "ack-id1";
+  private static final String ACK_ID2 = "ack-id2";
+  private static final String ACK_ID3 = "ack-id3";
+
+  private final Capture<Runnable> renewRunnable = Capture.newInstance();
+  private Clock clock;
+  private PubSub pubsub;
+  private ScheduledExecutorService executor;
+  private ExecutorFactory executorFactory;
+  private AckDeadlineRenewer ackDeadlineRenewer;
+  private ScheduledFuture scheduledFuture;
+
+  @Before
+  public void setUp() {
+    clock = EasyMock.createStrictMock(Clock.class);
```

Comment:
The more you mock the more susceptible you are to implementation details changes and in some cases you may over-mock and ignore potential issues.

I think it is reasonable to have some tests with timeouts and  you can use the `PusSub` last expected call as away indicate completion  (e.g. look at the first answer [here](http://stackoverflow.com/questions/16651578/using-easymock-in-multithreading)).

Just something to consider...